docs/README.md

```diff
@@ -0,0 +1,225 @@
+
+```proto
+message ProofSpec {
+  LeafOp    leaf_spec  = 1;
+  InnerSpec inner_spec = 2;
+  int32     max_depth  = 3;
+  int32     min_depth  = 4;
+  bool prehash_key_before_comparison = 5;
+}
+```
+
+where:
+
+- `max_depth` is the maximum number of inner nodes in the tree.
+- `min-depth` is the minimum number of inner nodes in the tree.
+- `prehash_key_before_comparison` is a flag that indicates whether to use the `prehash_key` hash operation specified in the `leaf_spec`. This is used to compare lexical ordering of keys for non-existence proofs.
+
+The `leaf_spec` is a [`LeafOp`](https://github.com/cosmos/ics23/blob/go/v0.10.0/proto/cosmos/ics23/v1/proofs.proto#L96-L120). The `LeafOp` type specifies the internal transformation from the original key/value pair of the leaf node into a hash. It specifies any prefix that should prepended and the hash operations to may transform the key and value, among other things:
+
+```proto
+message LeafOp {
+  HashOp   hash          = 1;
+  HashOp   prehash_key   = 2;
+  HashOp   prehash_value = 3;
+  LengthOp length        = 4;
+  bytes    prefix = 5;
+}
+```
+
+where:
+
+- `hash` is [hash operation](https://github.com/cosmos/ics23/blob/go/v0.10.0/proto/cosmos/ics23/v1/proofs.proto#L7-L16) that is applied to the result of `(prefix || length(prehash_key(key)) || length(prehash_value(value))`, where the `||` operator is concatenation of binary data and `key` and `value` are the key and value of the leaf node. 
+- `prehash_key` is the hash operation that is applied to the key of the leaf.
+- `prehash_value` is the hash operation that is applied to the value of the leaf.
+- `length` is the [length encoding operation](https://github.com/cosmos/ics23/blob/go/v0.10.0/proto/cosmos/ics23/v1/proofs.proto#L18-L43) that determines the length of the input and returns it prepended to the data.
+- `prefix` is a fixed set of bytes that may optionally be prepended to differentiate a leaf node from an inner node.
+
+And `inner_spec` is an [`InnerSpec`](https://github.com/cosmos/ics23/blob/go/v0.10.0/proto/cosmos/ics23/v1/proofs.proto#L172-L194). The `InnerSpec` specifies all store-specific structure information to determine if two proofs from a
+given store are neighbors.
+
+```proto
+message InnerSpec {
+  repeated int32 child_order       = 1;
+  int32          child_size        = 2;
+  int32          min_prefix_length = 3;
+  int32          max_prefix_length = 4;
+  bytes          empty_child       = 5;
+  HashOp         hash              = 6;
+}
+```
+
+where:
+
+- `child_order` is the ordering of the children node and counts from 0. For example, for an IAVL tree is [0, 1] (left, then right); for a merk tree is [0, 2, 1] (left, right, here).
+- `child_size`
+- `min_prefix_length`
+- `max_prefix_length`
```

Comment:
@AdityaSripal If you could explain me a bit what these field are exactly for, that would be great. I have a bit of hard time understanding these ones.